Packs/HackerOne/Integrations/HackerOne/HackerOne.py

```diff
@@ -0,0 +1,677 @@
+        page = LOWER_BOUND
+
+    return page, limit
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises:
+     exceptions if something goes wrong.
+
+    Args:
+        client: client to use
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+
+    if demisto.params().get('isFetch'):
+        fetch_incidents(client, {})
+    else:
+        client.program_list(params={"page[size]": 1})
+
+    return 'ok'
+
+
+def fetch_incidents(client: Client, last_run: dict, ) -> Tuple[dict, list]:
+    """Fetches incidents from HackerOne.
+
+    :type client: ``Client``
+    :param client: client to use
+
+    :type last_run: ``Dict[str, str]``
+    :param last_run: Last run returned by function demisto.getLastRun
+
+    :type params: ``Dict[str, str]``
+    :param params: Arguments for fetch-incident.
+
+    :rtype: ``Tuple``
+    :return: tuple of dictionary of next run and list of fetched incidents
+    """
+    validate_fetch_incidents_parameters(client.max_fetch, client.program_handle, client.filters)
+    fetch_params = prepare_fetch_incidents_parameters(client.max_fetch, client.first_fetch, client.program_handle,
+                                                      client.severity, client.state, client.filters)
+    first_fetch = fetch_params["filter[created_at__gt]"]
+
+    fetch_params["page[number]"] = last_run.get("next_page", 1)
+    fetch_params["filter[created_at__gt]"] = last_run.get("next_created_at", first_fetch)  # type: ignore
+
+    response = client.report_list(params=fetch_params)
+
+    results = response.get('data', [])
+    next_run = last_run
+    if not results:
+        return next_run, []
+
+    created_at_last_report = results[-1].get("attributes", {}).get("created_at")
+
+    if created_at_last_report == fetch_params["filter[created_at__gt]"]:
+        next_run = {
+            "next_page": fetch_params["page[number]"] + 1,
+            "next_created_at": fetch_params["filter[created_at__gt]"]
+        }
+    else:
+        next_run = {
+            "next_page": 1,
+            "next_created_at": created_at_last_report
+        }
+
+    report_ids = last_run.get("report_ids", [])
+    new_report_ids = []
+    incidents = []
+    for result in results:
+        if result.get("id") in report_ids:
+            continue
+        new_report_ids.append(result.get("id"))
+        incidents.append({
+            'name': result.get('attributes', {}).get('title', ''),
+            'occurred': result.get('attributes', {}).get('created_at'),
+            'rawJSON': json.dumps(result)
+        })
+
+    if created_at_last_report == fetch_params["filter[created_at__gt]"]:
+        next_run["report_ids"] = report_ids + new_report_ids
+    else:
+        next_run["report_ids"] = new_report_ids
```

Comment:
A bit of re-arrangment:
```suggestion
    previous_report_ids = last_run.get("report_ids", [])
    new_report_ids = []
    incidents = []
    for result in results:
        if result.get("id") not in previous_report_ids:
            new_report_ids.append(result.get("id"))
            incidents.append({
                'name': result.get('attributes', {}).get('title', ''),
                'occurred': result.get('attributes', {}).get('created_at'),
                'rawJSON': json.dumps(result)
            })
       
    next_page = 1,
    next_report_ids = new_report_ids
    created_at_last_report = results[-1].get("attributes", {}).get("created_at")
    if created_at_last_report == time_to_fetch:
        next_report_ids =  previous_report_ids + new_report_ids
        next_page: page_to_fetch + 1,

     next_run = {
            "next_page": next_page,
            "next_created_at": created_at_last_report,
            "report_ids": next_report_ids
        }
```